openassessment/workflow/models.py

```diff
@@ -239,7 +240,8 @@ def get_score(self, assessment_requirements, step_for_name):
                     else:
                         requirements = assessment_requirements.get(assessment_step_name, {})
                     score = get_score_func(self.submission_uuid, requirements)
-                    break
+                    if score:
+                        break
```

Comment:
`get_score` of existing apis can return `None` if the step is not complete. For staff assessment, `get_score` will also return `None` if the step is optional and staff have not accessed it. By making this change, the score of the next step will be used.